api/providers/ClientProvider.cs

```diff
@@ -0,0 +1,71 @@
+using DTO.Client;
+using FluentValidation;
+using Microsoft.EntityFrameworkCore;
+
+public class ClientProvider : BaseProvider<Client>
+{
+    private readonly IValidator<Client> _clientValidator;
+    private readonly ContactProvider _contactProvider;
+    private readonly AddressProvider _addressProvider;
+
+    public ClientProvider(AppDbContext db, IValidator<Client> clientValidator, ContactProvider contactProvider, AddressProvider addressProvider)
+        : base(db)
+    {
+        _clientValidator = clientValidator;
+        _contactProvider = contactProvider;
+        _addressProvider = addressProvider;
+    }
+
+
+    public override Client? Create(BaseDTO createValues)
+    {
+        ClientRequest request = createValues as ClientRequest ?? throw new ApiFlowException("Could not process create client request. Save new client failed.");
+        
+        if (request.ContactId == null && request.Contact == null) throw new ApiFlowException("Either contact_id or contact fields must be filled.");
+        if (request.AddressId == null && request.Address == null) throw new ApiFlowException("Either address_id or address fields must be filled.");
```

Comment:
Dit is niet nodig want eigenlijk gaat de validator dat opakken dit is nu extra code